.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,61 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Run tests from tests/
+
+on:
+  push:
+    branches: [ "master", "*" ]
+  pull_request:
+    branches: [ "master" ]
+
+  workflow_dispatch:
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install system dependencies
+        run: sudo apt install libdbus-1-dev libolm-dev
+
+      - name: Upgrade PIP
+        run: python -m pip install --upgrade pip
+
+      - name: Install requirements for Matrix Commander
+        run: pip install -r requirements.txt
+
+      - name: Source install of Matrix Commander
+        run: pip install .
+
+      - name: Log in to matrix server
+        env:
+          MATRIX_HOMESERVER: ${{ secrets.MATRIX_HOMESERVER }}
+          MATRIX_ROOM: ${{ secrets.MATRIX_ROOM }}
+          MATRIX_USERNAME: ${{ secrets.MATRIX_USERNAME }}
+          MATRIX_PASSWORD: ${{ secrets.MATRIX_PASSWORD }}
+        run: matrix-commander --login password --user-login "${MATRIX_USERNAME}" --password "${MATRIX_PASSWORD}" --homeserver "${MATRIX_HOMESERVER}" --device "ci" --room-default "${MATRIX_ROOM}" -m "matrix-commander posted from CI (${GITHUB_REPOSITORY} / ${GITHUB_REF})"
+
+      - name: Test events
+        run: ./tests/test-event.sh
+
+      - name: Test delete event
+        run:  ./tests/test-delete.sh
+
+      - name: Test REST
+        run:  ./tests/test-rest.sh
+
+      - name: Test set/get
+        run: ./tests/test-setget.sh
+
+      # - name: Test send
+      #   run: python ./tests/test-send.py
```

Comment:
For the time being I would add a comment line like "This test case is currently not working, but it has not yet been investigated why it fails in the workflow while working locally." Just to document the current situation.